### PR TITLE
eol_last: error if file ends with multiple newlines

### PR DESCRIPTION
**Pull request by charlierudolph**
@swang

### CODE REVIEW

**Comment by ervumlens** [comment]
Hi, quick question from the peanut gallery: is the problem with two or more trailing lines essentially aesthetic or is there a technical concern with it on the order of having no trailing lines at all?

**Comment by charlierudolph** [comment]
So the rule here has the description: `Checks that the file ends with a single newline`. Previously the rule was enforcing at least one. This updates make it enforce exactly one.

To me this is all aesthetics. multiple newlines at the end of a file is useless. A single newline is wanted for this [reason](http://stackoverflow.com/questions/729692/why-should-files-end-with-a-newline)

**Comment by ervumlens** [comment]
Roger that. I figured the change was to conform to the description ("single means *single!*"). Glad to know there isn't some obscure technical reason for me to worry about multiples. :)

File: src/rules/eol_last.coffee [hunk 1]
```diff
@@ -10,4 +10,11 @@ module.exports = class EOLLast
     lintLine: (line, lineApi) ->
         return null unless lineApi.isLastLine()
 
-        return true if line.length
+        isNewline = line.length is 0
+
+        previousIsNewline = if lineApi.lineCount > 1
+            lineApi.lines[lineApi.lineNumber - 1].length is 0
+        else
+            no
```
Comment (swang) [inline src/rules/eol_last.coffee]: can you make this false. i dont think we use no at all anywhere in the coffeelint code.

File: src/rules/eol_last.coffee [hunk 2]
```diff
@@ -10,4 +10,11 @@ module.exports = class EOLLast
...
+
+        return true unless isNewline && !previousIsNewline
```
Comment (swang) [inline src/rules/eol_last.coffee]: nit: can you use `and`, and `not`?

**Comment by swang** [comment]
Looks good, I'm on vacation right now on a fly-day. Be back to the States on Saturday. I'll prob make a new release then. Didn't want to do it before I went on vacation and have something go wrong. Anyways, Thanks!

File: src/rules/eol_last.coffee (hunk 1 above)
Comment (charlierudolph) [inline src/rules/eol_last.coffee]: fixed. Seems like you should be using a coffeelint rule for this

File: src/rules/eol_last.coffee (hunk 2 above)
Comment (charlierudolph) [inline src/rules/eol_last.coffee]: Same as above

File: src/rules/eol_last.coffee (hunk 1 above)
Comment (swang) [inline src/rules/eol_last.coffee]: I was going to wait until I decided what to do for a 2.0 version. But I think what will end up happening is 2.0 will enable a lot of rules I wanted on by default but didn't want to break semver.

**Pull request merged**